File: mpp_proiect_java/TriatlonServices/src/main/java/triatlon/services/ITriatlonServices.java
package triatlon.services;

import domain.*;

import java.util.List;

public interface ITriatlonServices {
    void login(Arbitru arbitru, ITriatlonObserver client) throws TriatlonException;
    void sendRezultat(Rezultat rezultat) throws TriatlonException;
    void logout(Arbitru arbitru, ITriatlonObserver client) throws TriatlonException;
    Arbitru getArbitrubyName(String name) throws TriatlonException;
    List<ParticipantDTO> getParticipantiDTO(Arbitru arbitru) throws TriatlonException;
    Proba getProbaArbitrubyId(long id) throws TriatlonException;
    List<Rezultat> filterbyProba(Proba proba) throws  TriatlonException;
    Participant findParticipantByNumePrenume(String firstName,String lastName) throws TriatlonException;

}

File: mpp_proiect_CSharp/triatlon/services/ITriatlonServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using triatlon.domain;
namespace services
{
   

    public interface ITriatlonServices

    {
        void login(Arbitru user, ITriatlonObserver client);
        void sendRezult(Rezultat rezultat);
        void logout(Arbitru user, ITriatlonObserver client);
        ParticipantDTO[] getParticipanti(Arbitru user);

        Arbitru getArbitruLogat(string username);

        Proba getProbaByArbitru(long id_proba);

        Participant findParticipantbyfirstlastName(ParticipantDTO part);

        Rezultat[] filterbyProba(Proba proba);

    }
}
